### DISCUSSION TITLE
Is it possible to have inline interactive node view?

### DISCUSSION THREAD

**Discussion by AkshayPathak** [ueberdosis/tiptap#3728] [General Questions]
Hi there, thanks for the fantastic library. 

I am looking at https://tiptap.dev/guide/node-views, but when implemented with a React component, I am not able to inline it. 

I am trying to create a drop down menu with text surrounding it such as the [Mention](https://tiptap.dev/api/nodes/mention#suggestion) extension.

Is this possible?

**Comment by briward** [comment]
I have also been unable to create an inline node that renders a NodeView either.

**Reply by litingyes** [reply]
I remember that all nodes support nodeViews

**Comment by MarkusTrb** [comment]
Spent too many hours on this but got to a working solution finally!

Add span as the contentDOMElementTag for ReactNodeViewRenderer:
`ReactNodeViewRenderer(ReactElement, { contentDOMElementTag: "span" });`

And make sure the NodeViewContent also is set to span like this:
`<NodeViewContent as="span" />`

**Reply by nperez0111** [reply]
I thought it would do that automatically if the node was specified as inline: true
